### DISCUSSION TITLE
REST API login help

### DISCUSSION THREAD

**Discussion by eldados** [Budibase/budibase#3804] [Help]
Hi,
Is there  way to connect to an external API using API key as well as basic login?
I can add and X-Application-Key in the header, but can't see a way to add user and password.
Cheers

**Comment by mjashanks** [comment]
It really depends on what your API supports. But the web standard for this would be via the header 
`Autorization: Basic <credentials>`

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

... but your API might not support this.

**Reply by barry-fisher317** [reply]
Any progress on the Oauth2 authentication?  That likely unlocks Twilio and Salesforce among others.  Will it make the December release?

**Reply by mjashanks** [reply]
So, the REST API updates will allow you to set up an initial authentication call to fetch a token. All subsequent requests can use this token.

However, in this version, you would have to use a hard-coded service account for that initial authentication call. I.e. there would be no personal login OAuth flow. That's a bigger project, for later.

We'll release the next update in early January, as we will take some time off.

**Comment by mielune** [comment]
Hi,

I would like to call an API behing an oauth2 gateway but I don't find any way to generate a token based on client_cred with client_id and client_secret and then use them into the authorization header bearer.
Have you an idea about any feature on it ?

KR
